clang/include/clang/Sema/Sema.h

```diff
@@ -13142,9 +13144,9 @@ void Sema::addIntelSYCLTripleArgFunctionAttr(Decl *D,
       !ZDimExpr->isValueDependent()) {
 
     // Save ConstantExpr in semantic attribute
-    XDimExpr = checkMaxWorkSizeAttrExpr(*this, CI, XDimExpr);
-    YDimExpr = checkMaxWorkSizeAttrExpr(*this, CI, YDimExpr);
-    ZDimExpr = checkMaxWorkSizeAttrExpr(*this, CI, ZDimExpr);
+    XDimExpr = checkWorkSizeAttrExpr(*this, CI, XDimExpr);
```

Comment:
Isn't this just going to recurse into this function infinitely?

Reply:
This was added as part of refactoring work here https://github.com/intel/llvm/pull/3169. No change  on this PR.